backend/src/main/scala/cromwell/backend/wdl/WdlFileMapper.scala

```diff
@@ -0,0 +1,28 @@
+package cromwell.backend.wdl
+
+import lenthall.util.TryUtil
+import wdl4s.values.{WdlArray, WdlFile, WdlMap, WdlValue}
+
+import scala.util.{Success, Try}
+
+object WdlFileMapper {
+  def mapWdlFiles(mapper: (WdlFile => WdlFile))(wdlValue: WdlValue): Try[WdlValue] = {
+    wdlValue match {
+      case file: WdlFile => Try(mapper(file))
+      case array: WdlArray =>
+        val mappedArray = array.value map mapWdlFiles(mapper)
+        TryUtil.sequence(mappedArray) map {
+          WdlArray(array.wdlType, _)
+        }
+      case map: WdlMap =>
+        val mappedMap = map.value map {
+          // TODO: TryUtil.sequenceMap doesn't currently handle keys of Try[WdlValue], so for now calling .get here
+          case (key, value) => mapWdlFiles(mapper)(key).get -> mapWdlFiles(mapper)(value)
```

Comment:
I'll write my own `TryUtil`-ish method for now, and it can go down into lenthall later.